### DISCUSSION TITLE
ENERGY_Total 37% too high (calibrated, power factor 1.0)

### DISCUSSION THREAD

**Discussion by gitisgreat2023** [arendst/Tasmota#20926] [Q&A]
### PROBLEM DESCRIPTION
Power calibration done as documented [here](https://tasmota.github.io/docs/Power-Monitoring-Calibration/). Nevertheless the energy counter increases about 37% too fast, when integrating the power over time, during charging.

Power calibration was done, the power factor is 1.00 during the test. Power, voltage and current are also matching during the test (about 1 kW, 232 volt and 4.3 amps).

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): TASMOTA Steckdose 16A, CURRYSMARTER V1.1 
- [x] Tasmota binary firmware version number used: 13.4.0
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
12:17:44.011 CMD: backlog template
12:17:44.090 MQT: stat/huawei_plug3/RESULT = {"NAME":"R4850G2","GPIO":[0,0,0,2624,32,320,0,0,224,2720,2656,0,0,0],"FLAG":0,"BASE":18}
12:17:45.253 MQT: tele/huawei_plug3/STATE = {"Time":"2024-03-11T12:17:45","Uptime":"0T00:27:14","UptimeSec":1634,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"XXX","BSSId":"XXX","Channel":13,"Mode":"11n","RSSI":36,"Signal":-82,"LinkCount":1,"Downtime":"0T00:00:09"}}
12:17:45.263 MQT: tele/huawei_plug3/SENSOR = {"Time":"2024-03-11T12:17:45","ENERGY":{"TotalStartTime":"2024-03-07T07:22:05","Total":21.009,"Yesterday":2.006,"Today":3.820,"Period":3,"Power":1062,"ApparentPower":1062,"ReactivePower":0,"Factor":1.00,"Voltage":232.10,"Current":4.575}}

12:19:10.978 MQT: stat/huawei_plug3/RESULT = {"Module":{"0":"R4850G2"}}

12:19:46.259 CMD: GPIO 255
12:19:46.271 MQT: stat/huawei_plug3/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"2624":"HLWBL SEL_i"},"GPIO4":{"32":"Button1"},"GPIO5":{"320":"Led_i1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"2720":"BL0937 CF"},"GPIO14":{"2656":"HLWBL CF1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
STATUS11 = {"StatusSTS":{"Time":"2024-03-11T12:14:02","Uptime":"0T00:23:31","UptimeSec":1411,"Heap":23,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"XXX","BSSId":"XXX","Channel":13,"Mode":"11n","RSSI":34,"Signal":-83,"LinkCount":1,"Downtime":"0T00:00:09"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
Calibrate power plug.
Over 1 hour have a constant load of about 1 kW.
Calculate the increase of ENERGY_Total in kWh over the time period and compare with the expected value (power times time duration in kWh)

### EXPECTED BEHAVIOUR
ENERGY_Total should increase as expected by the average power over a time period.,

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
The current did not change during the whole charging process? That's not usual.

**Reply by gitisgreat2023** [reply]
From what do you conclude the current didn't change during the whole process? It did... 
How did you come to that conclusion?

**Reply by Jason2866** [reply]
No log of any kind provided. So i assumed the usual charging behaviour.

**Reply by gitisgreat2023** [reply]
Okay, I'll get tomorrow logs over an hour which shows the too fast increase in energy.
During the hour it was relatively stable, the energy just increased too fast.

I just don't understand why, if power, voltage and current are all consistent with each other and the power factor is 1, that the energy increases too fast. The curve of the energy increase is just linearly going up, so the slope is wrong. That kind of exclude single spike events (or it would have to be many data artefacts all the time).

**Reply by gitisgreat2023** [reply]
Ok, same mismatch observed, stayed constant at 1.37x too hight (37%). I bought other plugs, lets see whether the issue is gone. I'll use these buggy ones with a correction factor of for example 1.37.

**Comment by arendst** [comment]
I guess you're using a wrong template.

Changing GPIO13 from BL0937_CF to HLW8012_CF most likely fixes your issue.

**Reply by gitisgreat2023** [reply]
Cool, I'll try that!
Can one find out, without opening (breaking the glued parts) the plug which hardware is in, such that such wrong configs one can find quickly oneself?
